test/unit_test/baseline_cb_test.cc

```diff
@@ -0,0 +1,143 @@
+#ifndef STATIC_LINK_VW
+#define BOOST_TEST_DYN_LINK
+#include <ios>
+#include <sstream>
+#include "reductions_fwd.h"
+#endif
+
+#include <boost/test/unit_test.hpp>
+#include <boost/test/test_tools.hpp>
+
+#include "test_common.h"
+
+#include <vector>
+
+#include "vw.h"
+
+void make_example(multi_ex& examples, vw& vw, int arm, float* costs, float* probs)
+{
+  examples.push_back(VW::read_example(vw, "shared | shared_f"));
+  for(int i = 0; i < 4; ++i) {
+    std::stringstream str;
+    if(i == arm) {
+      str << "0:" << std::fixed << costs[i] << ":" << probs[i] << " ";
+    }
+    str << "| " << "arm_" << i;
+    examples.push_back(VW::read_example(vw, str.str().c_str()));
+  }
+}
+
+int sample(int size, const float *probs, float s)
+{
+  for(int i = 0; i < size; ++i) {
+    if(s <= probs[i]) {
+      return i;
+    }
+    s -= probs[i];
+  }
+  return 0; //error
+}
+
+
+int get_int_metric(const VW::metric_sink& metrics, const std::string& metric_name)
```

Comment:
Should we try to avoid having almost the same util code duplicated across tests? https://github.com/VowpalWabbit/vowpal_wabbit/blob/6670fc8db7cc44e65f84e5538c9bf285b204dd1c/test/unit_test/automl_test.cc#L25

Reply:
Although its just a find_if, but makes me think that these sort of utility function might even be useful as part of metric_sink. Right now its pretty barebones, just a collection of lists so not exactly sure if it worth it

Reply:
metric_sink is not a great API as far as partial usage.
My preference would be something like dotnet's perf counters.
IE, pseudo-code 
```
vw::perf_counter<int> is_baseline = all->get_counter<int>("baseline_cb_baseline_in_use");
///the test code.
BOOST_CHECK_EQUAL(is_baseline.sample(), 0);
```